Assets/Scripts/ButtonClick.cs

```diff
@@ -0,0 +1,21 @@
+﻿using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.Events;
+
+[AddComponentMenu("UI/ButtonWithRightclick", 30)]
+public class ButtonClick : MonoBehaviour, IPointerClickHandler
+{
+    public UnityEvent leftClick;
+    public UnityEvent middleClick;
+    public UnityEvent rightClick;
```

Comment:
```suggestion
    public UnityEvent leftClick { get; set; }
    public UnityEvent middleClick { get; set; }
    public UnityEvent rightClick { get; set; }
```

Reply:
Why use a public Attribute instead of Getter/Setter?

Reply:
Because Properties cannot be shown in the inspector - and that is what the Events are for

Reply:
Unlucky